scripts/core/editor3/service.js

```diff
@@ -95,15 +95,31 @@ export class EditorService {
     /**
      * @ngdoc method
      * @name editor3#setSettings
-     * @param {Object} findReplace The new search criteria. An object containing the keys
-     * caseSensitive (boolean) and diff (object having one key that is the pattern).
-     * @description Updates the search criteria.
+     * @param {Object} The setting can be findreplace or spellcheck.
+     * @description Updates the settings for editor3.
      */
-    setSettings({findreplace}) {
+    setSettings(setting) {
         if (!ok()) {
             return;
         }
 
+        if ('findreplace' in setting) {
```

Comment:
We've not really used this syntax much in the codebase. I think this has been used more:
```js
if (typeof findreplace !== 'undefined')
```
Or alternatively:
```js
if (_.isDefined(findreplace))
```